packages/salesforcedx-sobjects-faux-generator/src/generator/declarationGenerator.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ChildRelationship, Field, SObject } from '../types';
+
+export const MODIFIER = 'global';
+export interface FieldDeclaration {
+  modifier: string;
+  type: string;
+  name: string;
+  comment?: string;
+}
+
+export interface SObjectDefinition {
+  name: string;
+  fields: FieldDeclaration[];
+}
+
+export class DeclarationGenerator {
```

Comment:
Eventually we need to switch the main entry point to `salesforcedx-sobject-faux-generator` to not be `FauxClassGenerator` and be something that ties/calls all the potential format outputs. Do you think this could be it or would we have a different class/code calling this, FauxClassGenerator and TypingsGenerator ?

Reply:
I'm leaning toward a different class as the orchestrator - DeclarationGenerator would just convert the data and FauxClassGenerator would just create the CLS output.

Reply:
Sounds good 👍

Reply:
Re-surfacing this thread, is it correct to assume this would happen on a follow up PR given `FauxClassGenerator` is still the entry point for sobject creation ?